Tpetra: compile failure with SYCL backend

## Bug Report
@trilinos/tpetra @tjfulle 

On Aurora, I'm seeing these errors, due to #14545.  Googling indicates that printing within a kernel requires some special handling with SYCL.

```
25936 In file included from /lus/flare/projects/kokkos_math/trilinos-performance-testing/build/packages/tpetra/core/src/Tpetra_CrsGraph_INT_LONG_LONG_SYCL.cpp:35:
25937 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7222:29: error: SYCL kernel cannot call a variadic function
25938  7222 |         if (srcGid == GINV) CRSGRAPH_INNER_ABORT(__LINE__);
25939       |                             ^
25940 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7207:52: note: expanded from macro 'CRSGRAPH_INNER_ABORT'
25941  7207 |     printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin); \
25942       |                                                    ^
25943 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/kokkos/core/src/SYCL/Kokkos_SYCL_ParallelFor_Range.hpp:56:9: note: called by 'operator()'
25944    56 |         m_functor_wrapper.get_functor()(shifted_id);
25945       |         ^
25946 /opt/aurora/24.180.3/updates/oneapi/compiler/eng-20240629/bin/compiler/../../include/sycl/handler.hpp:1648:5: note: called by 'kernel_parallel_for<Kokkos::Impl::FunctorWrapperRangePolicyParallelForCustom<Kokkos::Impl::SYCLFunctionWrapper<(lamb
25947  1648 |     KernelFunc(detail::Builder::getElement(detail::declptr<ElementType>()));
25948       |     ^
25949 In file included from /lus/flare/projects/kokkos_math/trilinos-performance-testing/build/packages/tpetra/core/src/Tpetra_CrsGraph_INT_LONG_LONG_SYCL.cpp:35:
25950 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7225:34: error: SYCL kernel cannot call a variadic function
25951  7225 |         if (tgtLocalRow == LINV) CRSGRAPH_INNER_ABORT(__LINE__);
25952       |                                  ^
25953 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7207:52: note: expanded from macro 'CRSGRAPH_INNER_ABORT'
25954  7207 |     printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin); \
25955       |                                                    ^
25956 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7226:36: error: SYCL kernel cannot call a variadic function
25957  7226 |         if (tgtLocalRow != tgtLid) CRSGRAPH_INNER_ABORT(__LINE__);
25958       |                                    ^
25959 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7207:52: note: expanded from macro 'CRSGRAPH_INNER_ABORT'
25960  7207 |     printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin); \
25961       |                                                    ^
25962 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7247:27: error: SYCL kernel cannot call a variadic function
25963  7247 |           if (gi == GINV) CRSGRAPH_INNER_ABORT(__LINE__);
25964       |                           ^
25965 /lus/flare/projects/kokkos_math/trilinos-performance-testing/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:7207:52: note: expanded from macro 'CRSGRAPH_INNER_ABORT'
25966  7207 |     printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin); \
25967       |                                                    ^
25968 2 warnings and 4 errors generated.
```

--- a/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp
+++ b/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp
@@ -13,6 +13,10 @@
 /// \file Tpetra_CrsGraph_def.hpp
 /// \brief Definition of the Tpetra::CrsGraph class
 
+#ifdef KOKKOS_ENABLE_SYCL
+#include <sycl/sycl.hpp>
+#endif
+
 #include "Tpetra_Details_Behavior.hpp"
 #include "Tpetra_Details_computeOffsets.hpp"
 #include "Tpetra_Details_copyOffsets.hpp"
@@ -7202,11 +7206,19 @@ void CrsGraph<LocalOrdinal, GlobalOrdinal, Node>::insertGlobalIndicesDevice(
 #undef CRSGRAPH_INNER_ABORT
 #endif
 
+#ifdef KOKKOS_ENABLE_SYCL
+#define CRSGRAPH_INNER_ABORT(lin)                                                      \
+  do {                                                                                 \
+    sycl::ext::oneapi::experimental::printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin); \
+    Kokkos::abort("error");                                                            \
+  } while (0)
+#else
 #define CRSGRAPH_INNER_ABORT(lin)                     \
   do {                                                \
     printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin); \
     Kokkos::abort("error");                           \
   } while (0)
+#endif
 
   Kokkos::parallel_for(
       "Tpetra_CrsGraph::copyAndPermuteNew",

